flytekit/core/inference.py

```diff
@@ -0,0 +1,182 @@
+"""
+=========
+Inference
+=========
+
+.. currentmodule:: flytekit.core.inference
+
+This module includes inference subclasses that extend the `ModelInferenceTemplate`.
+
+.. autosummary::
+   :nosignatures:
+   :template: custom.rst
+   :toctree: generated/
+
+   NIM
+"""
+
+from typing import Optional
+
+from .utils import ModelInferenceTemplate
+
+
+class NIM(ModelInferenceTemplate):
+    def __init__(
+        self,
+        image: str = "nvcr.io/nim/meta/llama3-8b-instruct:1.0.0",
+        health_endpoint: str = "v1/health/ready",
+        port: int = 8000,
+        cpu: int = 1,
+        gpu: int = 1,
+        mem: str = "20Gi",
+        shm_size: str = "16Gi",
+        # kubernetes secrets
+        ngc_image_secret: Optional[str] = None,
+        ngc_secret_key: Optional[str] = None,
```

Comment:
Do these have to be k8s secrets? Why can't we use Flyte secrets?

Reply:
`ngc_secret_key` must be the name of the Flyte secret, whereas `ngc_image_secret` must be a Kubernetes secret.